presto-cache/src/main/java/com/facebook/presto/cache/alluxio/AlluxioCachingHdfsFileInputStream.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cache.alluxio;
+
+import alluxio.client.file.FileInStream;
+import alluxio.exception.ExceptionMessage;
+import org.apache.hadoop.fs.PositionedReadable;
+import org.apache.hadoop.fs.Seekable;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class AlluxioCachingHdfsFileInputStream
+        extends InputStream
+        implements Seekable, PositionedReadable
+{
+    private final FileInStream inputStream;
+
+    public AlluxioCachingHdfsFileInputStream(FileInStream inputStream)
+    {
+        this.inputStream = inputStream;
+    }
+
+    @Override
+    public int read()
+            throws IOException
+    {
+        return inputStream.read();
+    }
+
+    @Override
+    public int read(long position, byte[] buffer, int offset, int length)
+            throws IOException
+    {
+        return inputStream.positionedRead(position, buffer, offset, length);
+    }
+
+    @Override
+    public void readFully(long position, byte[] buffer, int offset, int length)
+            throws IOException
+    {
+        int totalBytesRead = 0;
+        while (totalBytesRead < length) {
+            int bytesRead =
+                    read(position + totalBytesRead, buffer, offset + totalBytesRead, length - totalBytesRead);
```

Comment:
nit: maybe put them into one line? If the line is too long, we could put one argument per line.

Reply:
+1